src/components/InventoryV2/InventoryListByBatch.tsx

```diff
@@ -86,6 +92,56 @@ export default function InventoryListByBatch({ setReportData }: InventoryListByB
   });
   const [filters, setFilters] = useForm<InventoryFiltersType>({});
 
+  const dispatch = useAppDispatch();
+  const subLocations = useAppSelector(selectSubLocations);
+  const [subLocationNames, setSubLocationNames] = useState<string[]>([]);
+  const [nurseryIds, setNurseryIds] = useState<number[]>([]);
+  const query = useQuery();
+  useEffect(() => {
+    const querySubLocationNames = query
+      .get('subLocationName')
+      ?.trim()
+      ?.split(',')
+      ?.filter((s) => s !== '');
+    setSubLocationNames(querySubLocationNames ?? []);
+    const queryNurseryIds = query
+      .get('facilityId')
+      ?.trim()
+      ?.split(',')
+      ?.filter((n) => n !== '')
+      ?.map((id) => Number(id));
+    setNurseryIds(queryNurseryIds ?? []);
+  }, [query]);
+
+  useEffect(() => {
+    setFilters((f) => ({ ...f, facilityIds: nurseryIds }));
+  }, [nurseryIds, setFilters]);
+
+  useEffect(() => {
+    dispatch(requestSubLocations(filters.facilityIds ?? []));
+  }, [filters.facilityIds, dispatch]);
+
+  useEffect(() => {
+    const subLocationsIds = subLocationNames
+      .map((name) => subLocations?.find((sl) => sl.name === name)?.id)
+      .filter((id) => id !== undefined) as number[];
```

Comment:
You might be able to use a typeguard on the filter which would remove the need to cast.
```suggestion
      .filter((id): id is number => id !== undefined);
```